cmd/vela-worker/exec.go

```diff
@@ -93,63 +93,58 @@ func (w *Worker) exec(index int) error {
 		t = time.Duration(item.Repo.GetTimeout()) * time.Minute
 	}
 
-	// create a background context
-	ctx := context.Background()
+	// create a build context
+	buildCtx, buildDone := context.WithCancel(ctx)
+	defer buildDone()
 
 	// add to the background context with a timeout
 	// built in for ensuring a build doesn't run forever
-	ctx, timeout := context.WithTimeout(ctx, t)
+	timeoutCtx, timeout := context.WithTimeout(buildCtx, t)
 	defer timeout()
 
 	defer func() {
 		logger.Info("destroying build")
 		// destroy the build with the executor
-		err = _executor.DestroyBuild(context.Background())
+		// using buildCtx instead of timeoutCtx so that it happens after the timeout
+		err = _executor.DestroyBuild(buildCtx)
```

Comment:
```suggestion
		err = _executor.DestroyBuild(context.Background())
```
@jbrockopp said that:
> I think we'll want to be careful about passing the same context around everywhere.
> For example, specific to DestroyBuild() , we intentionally pass context.Background() instead of the shared context.
> The reason for that is because we want to guarantee all build resources are destroyed. If you pass the shared context then you run the risk of the build not getting destroyed.

Reply:
OK. I reverted this so that DestroyBuild still gets a background context. I added a comment about why.